src/api/artifact/abstractor/abstractor.go

```diff
@@ -109,10 +116,25 @@ func (a *abstractor) Abstract(ctx context.Context, artifact *artifact.Artifact)
 		return fmt.Errorf("unsupported manifest media type: %s", artifact.ManifestMediaType)
 	}
 
-	resolver, err := resolver.Get(artifact.MediaType)
-	if err != nil {
-		return err
+	resolver := resolver.Get(artifact.MediaType)
+	if resolver != nil {
+		artifact.Type = resolver.ArtifactType()
+		return resolver.Resolve(ctx, content, artifact)
+	}
+
+	// if got no resolver, try to parse the artifact type based on the media type
+	artifact.Type = parseArtifactType(artifact.MediaType)
+	// still can not get the artifact type, set it to unknown
+	if len(artifact.Type) == 0 {
+		artifact.Type = ArtifactTypeUnknown
+	}
+	return nil
+}
+
+func parseArtifactType(mediaType string) string {
+	strs := artifactTypeRegExp.FindStringSubmatch(mediaType)
```

Comment:
It would be slightly better if you let `parseArtifactType ` return unknown. And you should clearly print a log like `xxxx is not recognized returning unknown artifact type`